CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(UriParser VERSION 0.9.1 LANGUAGES C)
+
+file(GLOB INTERFACE_FILES
+        RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" CONFIGURE_DEPENDS
+        "include/uriparser/*.h")
+
+file(GLOB SOURCE_FILES
+        RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" CONFIGURE_DEPENDS
+        "src/*.[c|h]")
+
+add_library(uriparser ${INTERFACE_FILES} ${SOURCE_FILES})
+set_property(TARGET uriparser PROPERTY PUBLIC_HEADER "${INTERFACE_FILES}")
+target_include_directories(uriparser PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+target_include_directories(uriparser PUBLIC $<INSTALL_INTERFACE:include>)
```

Comment:
Do you have a good pointer on where to learn more about `$<BUILD_INTERFACE:...>` and `$<INSTALL_INTERFACE:...>`?

Reply:
These are called "Generator Expressions". We use a generator here because we want the target to have different public include paths when installed as a package (INSTALL) and when being built as a target of a larger CMAKE build tree (e.g. add_subdirectories or when running tests).

https://cmake.org/cmake/help/v3.13/manual/cmake-generator-expressions.7.html

Reply:
Thanks!